awscli/botocore/configprovider.py

```diff
@@ -561,3 +568,159 @@ def provide(self):
 
     def __repr__(self):
         return 'ConstantProvider(value=%s)' % self._value
+
+
+class ConfiguredEndpointProvider(BaseProvider):
```

Comment:
This class does not match up with what we have in the botocore PR: https://github.com/boto/botocore/pull/2973/files#diff-1818b8b628e5570486a0305e24ca6036a980fae9b2b73125b3298c7d02163313R862. Specifically I still see the `__repr__` and setting of `self._transformed_service_id`. It might be worth just copying this class over in its entirety to this PR.